sklearn/utils/extmath.py

```diff
@@ -275,8 +275,7 @@ def randomized_range_finder(
     # from float64 to float32 in asarray might not always be accepted as only
     # casts following type promotion rules are guarateed to work.
     # https://github.com/data-apis/array-api/issues/647
-    if is_array_api_compliant:
-        Q = xp.asarray(Q, device=device(A))
+    Q = xp.asarray(Q, device=device(A))
```

Comment:
I think we may still need the compliance check for use on sparse data (current test failures).

```suggestion
    if is_array_api_compliant:
        Q = xp.asarray(Q, device=device(A))
```